### DISCUSSION TITLE
How to properly decompile a workshop map?

### DISCUSSION THREAD

**Discussion by thecatontheflat** [SteamDatabase/ValveResourceFormat#770] [General] [answered]
Hi!

I am trying to decompile a workshop map. It seems that it's a VPK file that is bundled into another "root" VPK along with its related assets.
<img width="295" alt="image" src="https://github.com/ValveResourceFormat/ValveResourceFormat/assets/727581/7168d3f6-65dc-4486-ae7e-56db0fa72da5">
![image](https://github.com/ValveResourceFormat/ValveResourceFormat/assets/727581/84cc6960-3199-4c94-84fc-00c1faf727b1)

When I'm viewing it from within the "root" VPK, it has the textures (just some are missing, because apparently it references some assets from the CS2, but  for that I just need to place it in the corresponding CS2 folder, but let's ignore that for now).
![image](https://github.com/ValveResourceFormat/ValveResourceFormat/assets/727581/42779fb8-43b9-4b97-b1b4-ebb0a12e9b59)

Now when I decompile & export the map (like so):
![image](https://github.com/ValveResourceFormat/ValveResourceFormat/assets/727581/282153e1-394b-40ce-9589-b062da2f06eb)

and then open the decompiled VPK, it doesn't have any textures.
![image](https://github.com/ValveResourceFormat/ValveResourceFormat/assets/727581/9668aba7-e211-4bfd-bf01-663c9f6d2c05)

My guess is that I need to keep the folder structure, but I can't seem to figure the expected files layout.

Please help?

**Comment by xPaw** [comment, marked as answer]
Don't move or extract the inner vpk. If the workshop vpk is in the Steam workshop folder, it should correctly resolve all the dependencies.

For all other cases, that's what "search paths" in settings is for.

**Reply by thecatontheflat** [reply]
Thanks! I'll give it a try and will get back.

Is there a corresponding flag for adding search paths when using CLI decompiler?

**Reply by xPaw** [reply]
No, but it should correctly resolve from within the Steam folder, same as GUI.

PRs welcome to add a param to supply extra search paths.

**Reply by thecatontheflat** [reply]
I managed to get it work by decompiling the workshop VPK into the cs2 folder. I was worried it'd replace existing folders, but it just merged new files into the existing folders, so it seems to work nicely! Thanks!
